experimental/Schemes/SpecialTypes.jl

```diff
@@ -59,6 +59,17 @@ function generic_fraction(a::MPolyQuoLocalizedRingElem, U::PrincipalOpenSubset)
   return lifted_numerator(a)//lifted_denominator(a)
 end
 
+function is_dense(U::PrincipalOpenSubset)
+  return !is_zero_divisor(complement_equation(U))
+end
+
+function is_zero_divisor(f::Union{<:MPolyElem, <:MPolyLocalizedRingElem})
+  return iszero(f)
+end
```

Comment:
This is wrong for `(Z/4Z)[x, y]`.

Reply:
Oops... Yes, thanks for catching this. Either way: This should probably be implemented for the polynomial rings somewhere else, but I didn't find it anywhere.

Reply:
Maybe do a `coefficient_ring(f) isa Field || Hecke.NotImplemented()`.

Reply:
```
function is_zero_divisor(x::AbstractAlgebra.MPolyElem{T}) where T <: RingElement
```
is already in AA/src/MPoly.jl, so yall will have some difficulties overriding it generically.

Reply:
What do you meant with "We don't have such an implementation."?

Reply:
We only have a implementation for R[x] when R is a gcd domain.

Reply:
@HechtiDerLachs Did you actually try `is_zero_divisor(f)` for some multivariate polynomial?

Reply:
@tthsqe12: No, it is more annoying. `gcd` works for `nmod`, but
```
julia> R, = quo(ZZ, 4)
(Integers modulo 4, Map with section with the following data

Domain:
=======
Integer Ring

Codomain:
========
Integers modulo 4)

julia> Rxy, (x, y) = R["x", "y"]
(Multivariate Polynomial Ring in x, y over Integers modulo 4, nmod_mpoly[x, y])

julia> is_zero_divisor(2 * x)
ERROR: MethodError: no method matching content(::nmod_mpoly)
Closest candidates are:
  content(::AbstractAlgebra.Generic.MPoly{T}) where T<:RingElement at ~/.julia/packages/AbstractAlgebra/mgYkc/src/generic/MPoly.jl:3573
```
There needs to be an `MPolyElem` instead of `MPoly`.